src/makeInfinityAdder.js

```diff
@@ -1,10 +1,27 @@
 'use strict';
 
 /***
+ *
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      const sum = result;
+      result = 0;
+
+      return sum;
+    }
+
+    result += arg;
+
+    return adder;
```

Comment:
The current implementation of the adder function does not check if the argument is a number before adding it to the result. It's important to ensure that only numbers are added to prevent unexpected behavior. Consider adding a check to ensure that the argument is a number and throw an error or ignore non-number values.